src/Controllers/ChecksArrivalsAndDepartures.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Engelsystem\Controllers;
+
+use Carbon\Carbon;
+use DateTime;
+
+trait ChecksArrivalsAndDepartures
+{
+    protected function isArrivalDateValid(?string $arrival, ?string $departure): bool
+    {
+        $arrival_carbon = $this->toCarbon($arrival);
+        $departure_carbon = $this->toCarbon($departure);
+
+        if (is_null($arrival_carbon)) {
+            return false; // since required value
+        }
+
+        if (!is_null($departure_carbon) && $arrival_carbon->greaterThan($departure_carbon)) {
+            return false;
+        }
+
+        return !$this->isBeforeBuildup($arrival_carbon) && !$this->isAfterTeardown($arrival_carbon);
+    }
+
+    protected function isDepartureDateValid(?string $arrival, ?string $departure): bool
```

Comment:
I personally prefer having doc blocks for visual seperation but thats just a note from my side